### PR TITLE
Adjust sample

### PR DESCRIPTION
**Pull request by tom-osika**
Remove some /r characters that snuck into the jupyter notebooks. This was causing synchronization issues when editing.

Removed some files from the sample/ directory that are generated

Made a few adjustments to the Autopybind11Test.ipynb, like making the argument to os.path.join a list, fixing inconsistent usage of yaml vs yml, and typo in CMakeLists.txt

### CODE REVIEW

File: sample/AutoPyBind11Test.ipynb [hunk 1]
```diff
@@ -165,7 +150,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "visualize_binding_code(BINDING_CODE_DIR)"
+    "visualize_binding_code(os.path.join(binding_DEST, \"%s.cpp\" % cmake_library_name))"
```
Comment (tom-osika) [inline sample/AutoPyBind11Test.ipynb]: @johnwparent Can you confirm that this change is OK?

Comment (johnwparent) [reply sample/AutoPyBind11Test.ipynb]: Using MSVC this path won't work (which is why it is the way it is) but for the binder instance this change works. Was hoping to make this repo functionality for a local checkout but if it's causing errors w/ Binder then this change is fine.

Comment (tom-osika) [reply sample/AutoPyBind11Test.ipynb]: I was under the assumption that `os.path.join` was for that exact purpose? Or do you mean the fact that we're referring to the `.` directory here?

Comment (johnwparent) [reply sample/AutoPyBind11Test.ipynb]: Right, the `binding_DEST` variable refers to the root of the build repo. In MSVC the build output is then nested under a directory specifying the build configuration. The separator is not the issue here, rather the directory structure produced by MSVC.

Comment (tom-osika) [reply sample/AutoPyBind11Test.ipynb]: Good catch. Probably needs to be reverted then.

Comment (tom-osika) [reply sample/AutoPyBind11Test.ipynb]: See altered f5e945e38a5c50c3be4fdf1a3e4f85e082f3171c

**Comment by tom-osika** [comment]
These changes currently work for me using the simple example from APB. Would one of you give this a double check before merging?

**Review by johnwparent — APPROVED** [review]

**Comment by johnwparent** [comment]
Looks good to me. I'm ok with merging this (feel free to merge whenever you're ready) but if afterwards you could verify that the binder instance is functional, that would be greatly appreciated.

**Review by josephsnyder — APPROVED** [review]

**Pull request merged**